parquet/src/main/java/org/apache/iceberg/parquet/ParquetBloomRowGroupFilter.java

```diff
@@ -0,0 +1,325 @@
+    public <T> Boolean eq(BoundReference<T> ref, Literal<T> lit) {
+      int id = ref.fieldId();
+      if (!fieldsWithBloomFilter.contains(id)) { // no bloom filter
+        return ROWS_MIGHT_MATCH;
+      }
+
+      BloomFilter bloom = loadBloomFilter(id);
+      Type type = types.get(id);
+      T value = lit.value();
+      return shouldRead(parquetPrimitiveTypes.get(id), value, bloom, type);
+    }
+
+    @Override
+    public <T> Boolean notEq(BoundReference<T> ref, Literal<T> lit) {
+      // bloom filter is based on hash and cannot eliminate based on notEq
+      return ROWS_MIGHT_MATCH;
+    }
+
+    @Override
+    public <T> Boolean in(BoundReference<T> ref, Set<T> literalSet) {
+      int id = ref.fieldId();
+      if (!fieldsWithBloomFilter.contains(id)) { // no bloom filter
+        return ROWS_MIGHT_MATCH;
+      }
+      BloomFilter bloom = loadBloomFilter(id);
+      Type type = types.get(id);
+      for (T e : literalSet) {
+        if (shouldRead(parquetPrimitiveTypes.get(id), e, bloom, type)) {
+          return ROWS_MIGHT_MATCH;
+        }
+      }
+      return ROWS_CANNOT_MATCH;
+    }
+
+    @Override
+    public <T> Boolean notIn(BoundReference<T> ref, Set<T> literalSet) {
+      // bloom filter is based on hash and cannot eliminate based on notIn
+      return ROWS_MIGHT_MATCH;
+    }
+
+    @Override
+    public <T> Boolean startsWith(BoundReference<T> ref, Literal<T> lit) {
+      // bloom filter is based on hash and cannot eliminate based on startsWith
+      return ROWS_MIGHT_MATCH;
+    }
+
+    @Override
+    public <T> Boolean notStartsWith(BoundReference<T> ref, Literal<T> lit) {
+      // bloom filter is based on hash and cannot eliminate based on startsWith
+      return ROWS_MIGHT_MATCH;
+    }
+
+    private BloomFilter loadBloomFilter(int id) {
+      if (bloomCache.containsKey(id)) {
+        return bloomCache.get(id);
+      } else {
+        ColumnChunkMetaData columnChunkMetaData = columnMetaMap.get(id);
+        BloomFilter bloomFilter = bloomReader.readBloomFilter(columnChunkMetaData);
+        if (bloomFilter == null) {
+          throw new IllegalStateException("Failed to read required bloom filter for id: " + id);
+        } else {
+          bloomCache.put(id, bloomFilter);
+        }
+
+        return bloomFilter;
+      }
+    }
+
+    private <T> boolean shouldRead(PrimitiveType primitiveType, T value, BloomFilter bloom, Type type) {
+      long hashValue = 0;
+      switch (primitiveType.getPrimitiveTypeName()) {
+        case INT32:
+          switch (type.typeId()) {
+            case DECIMAL:
+              BigDecimal decimalValue = (BigDecimal) value;
+              hashValue = bloom.hash(decimalValue.unscaledValue().intValue());
+              return bloom.findHash(hashValue);
+            case INTEGER:
+            case DATE:
+              hashValue = bloom.hash(((Number) value).intValue());
+              return bloom.findHash(hashValue);
+          }
+        case INT64:
```

Comment:
Looks like this will fall through if the type ID is not handled? I think you need to add a `break;`.

Reply:
Will fix.